### PR TITLE
[kata/apps-symphony/M001/S09] feat(M001/S09): Conformance Sweep and Integration Polish

### PR DESCRIPTION
**Pull request by gannonh**
## What Changed
Closed the two spec §17 conformance test gaps, shipped a 425-line operator README, and confirmed 211 tests green + clippy clean — M001 milestone complete.

## Must-Haves
- `max_concurrent_agents_by_state` key normalization + invalid-value filtering is asserted by a dedicated test in `tests/workflow_config_tests.rs`
- Non-active state reconcile stop-without-cleanup path has an isolated test in `tests/orchestrator_tests.rs`
- `README.md` documents build, run, configuration, WORKFLOW.md schema, HTTP dashboard endpoints, SSH pool, and test invocation
- `cargo test` passes all tests (all prior 159 + 2 new = 161+)
- `cargo clippy -- -D warnings` exits zero
- `REQUIREMENTS.md` marks R013 validated with proof

## Tasks
- T01: Add conformance test gap closure
- T02: Write operator-grade README
- T03: Final verification, REQUIREMENTS.md update, and milestone close

<h3>Greptile Summary</h3>

This PR closes the final two spec §17 conformance gaps for the M001 milestone: it adds zero-value filtering to the `max_concurrent_agents_by_state` normalization pipeline in `config.rs`, backs both changes with dedicated integration tests, and replaces the internal agent-instruction guide in `AGENTS.md` with a 425-line operator README covering build, config schema, HTTP API, and SSH pool usage.

**Key changes:**
- `src/config.rs` — adds `.filter(|(_, v)| *v > 0)` before the `.to_lowercase()` map step, correctly discarding zero-value per-state caps as required by spec §17.1
- `tests/workflow_config_tests.rs` — `test_by_state_concurrency_normalization` asserts key normalization and zero-value filtering; missing a spaced-key round-trip case (e.g. `"In Progress"` → `"in progress"`) that matches real Linear state names
- `tests/orchestrator_tests.rs` — `test_reconcile_non_active_state_stops_run_without_cleanup` correctly validates that `release_issue` (not `mark_issue_terminal`) is called for issues in non-active, non-terminal states
- `AGENTS.md` — the closing "Development" section links to `AGENTS.md` itself, which is a self-referential circular reference since no separate `README.md` exists in `apps/symphony/`

<h3>Confidence Score: 4/5</h3>

- This PR is safe to merge; the code changes are small, correct, and well-tested — only a documentation self-reference and a minor test coverage gap need fixing.
- The production code change (one-line filter addition in config.rs) is logically correct and consistent with the existing normalization pattern. Both new tests verify their stated assertions. The two issues found are: (1) a self-referential link in AGENTS.md that has no functional impact, and (2) a missing test assertion for spaced keys that doesn't affect current behavior but represents a coverage gap for future regressions.
- apps/symphony/AGENTS.md (self-referential Development section), apps/symphony/tests/workflow_config_tests.rs (missing spaced-key normalization assertion)

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/symphony/src/config.rs | Adds `.filter(|(_, v)| *v > 0)` to the `max_concurrent_agents_by_state` pipeline — correctly removes zero-value entries per spec §17.1 before the lowercase normalization step; logic and ordering are sound. |
| apps/symphony/tests/orchestrator_tests.rs | New test `test_reconcile_non_active_state_stops_run_without_cleanup` correctly seeds the running map, triggers a tick with a non-active/non-terminal state ("In Review"), and asserts the issue is removed from `running` but not added to `completed`; test logic aligns with `release_issue` vs `mark_issue_terminal` semantics in the orchestrator. |
| apps/symphony/tests/workflow_config_tests.rs | New `test_by_state_concurrency_normalization` correctly tests uppercase-to-lowercase normalization and zero-value filtering, but omits a round-trip assertion for spaced keys (e.g. `"In Progress"` → `"in progress"`) that would catch future regressions on the most common real-world usage pattern. |
| apps/symphony/AGENTS.md | 425-line operator README replacing the agent instruction guide; comprehensive CLI, config, HTTP API, and SSH pool coverage, but the closing "Development" section self-references `AGENTS.md` (the same file), creating a circular link with no navigation value. |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A["WORKFLOW.md\nmax_concurrent_agents_by_state:\n  InProgress: 2\n  Review: 0\n  in_review: 3"] --> B["from_workflow()\nconfig.rs"]
    B --> C[".unwrap_or_default()\n(empty map if absent)"]
    C --> D[".filter — v > 0\nremove zero-value entries\nspec §17.1 NEW"]
    D --> E[".map — k.to_lowercase()\nnormalize keys to ASCII lowercase"]
    E --> F["HashMap result\ninprogress: 2\nin_review: 3"]
    F --> G["AgentConfig\n.max_concurrent_agents_by_state"]

    H["Issue reconcile\nstate = 'In Progress'"] --> I["normalize_issue_state\ntrim + to_ascii_lowercase"]
    I --> J["lookup key\n'in progress'"]
    J -->|"key found"| K["apply per-state cap"]
    J -->|"key not found"| L["use global\nmax_concurrent_agents"]

    style D fill:#c8e6c9,stroke:#388e3c
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/symphony/AGENTS.md
Line: 418-425

Comment:
**Self-referential Development section**

The "Development" section points to `AGENTS.md` — but this file *is* `AGENTS.md`. Since there is no separate `README.md` in `apps/symphony/`, the link resolves to the current document, creating a circular reference that provides no useful navigation.

Either remove the section entirely, or update it to point to a real file (e.g. a `README.md` that this PR description says was the goal of task T02, or at minimum change the link to an internal anchor such as `#commands`):

```suggestion
See the sections above for build, test, lint commands, and configuration details.

For architecture details and development guidelines:
- Module layout and Elixir reference: see the table in `AGENTS.md` (below the "Commands" section in the original developer guide)
- Hard rules for spec parity and implementation notes are tracked in `.kata/DECISIONS.md`
```

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/symphony/tests/workflow_config_tests.rs
Line: 471-509

Comment:
**Test key `InProgress` normalizes to `inprogress`, not `in progress`**

The test asserts that the config key `InProgress` normalizes to `"inprogress"` (no space). This is technically correct for key normalization (`.to_lowercase()` is applied), but it silently encodes a non-obvious behavior: a user who writes `InProgress: 3` expecting to cap the Linear state "In Progress" will get a dead key (`"inprogress"`) that never matches, because `normalize_issue_state("In Progress")` → `"in progress"` (space preserved).

The test would be more useful if it also exercised the spaced-key round-trip that mirrors real Linear state names:

```suggestion
fn test_by_state_concurrency_normalization() {
    // Build WORKFLOW.md with a mix of uppercase, zero-value, spaced, and valid entries.
    let mut file = NamedTempFile::new().unwrap();
    writeln!(
        file,
        "---\nagent:\n  max_concurrent_agents_by_state:\n    InProgress: 2\n    Review: 0\n    in_review: 3\n    In Progress: 4\n---\ntemplate"
    )
    .unwrap();

    let workflow = parse_workflow(file.path()).expect("should parse workflow");
    let config = from_workflow(&workflow.config).expect("should convert workflow to config");

    let by_state = &config.agent.max_concurrent_agents_by_state;

    // Uppercase key must be normalized to lowercase (note: spaces are preserved).
    assert_eq!(
        by_state.get("inprogress"),
        Some(&2u32),
        "InProgress key must be normalized to 'inprogress'"
    );
    // Original casing must not appear.
    assert!(
        !by_state.contains_key("InProgress"),
        "original uppercase key 'InProgress' must not survive normalization"
    );

    // Zero-value entry must be filtered out (spec §17.1: ignores invalid values).
    assert!(
        !by_state.contains_key("review"),
        "zero-value entry 'Review: 0' must be filtered (value 0 is invalid)"
    );

    // Valid lowercase entry must survive unchanged.
    assert_eq!(
        by_state.get("in_review"),
        Some(&3u32),
        "valid lowercase entry 'in_review: 3' must be preserved"
    );

    // Spaced key matching a typical Linear state (e.g. "In Progress") must be
    // case-folded but keep its space, since normalize_issue_state also preserves spaces.
    assert_eq!(
        by_state.get("in progress"),
        Some(&4u32),
        "'In Progress: 4' must normalize to 'in progress' (space preserved)"
    );
}
```

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Last reviewed commit: ["style: cargo fmt orc..."](https://github.com/gannonh/kata/commit/08577fd45b954efb8e273c86aec181f9cb3988b3)</sub>

> Greptile also left **2 inline comments** on this PR.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed agent configuration to filter zero-valued concurrency limits
  * Fixed orchestrator state handling for non-active, non-terminal states

* **Tests**
  * Added test coverage for configuration key normalization and zero-value filtering
  * Added test for orchestrator behavior with edge-case state transitions

* **Chores**
  * Added `.env` file to gitignore for security

### CODE REVIEW

File: apps/symphony/tests/workflow_config_tests.rs [hunk 1]
```diff
@@ -466,3 +466,44 @@ async fn test_workflow_store_force_reload_reports_error_on_invalid_workflow() {
...
+        "zero-value entry 'Review: 0' must be filtered (value 0 is invalid)"
+    );
+
+    // Valid lowercase entry must survive unchanged.
+    assert_eq!(
+        by_state.get("in_review"),
+        Some(&3u32),
+        "valid lowercase entry 'in_review: 3' must be preserved"
+    );
+}
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/symphony/tests/workflow_config_tests.rs:517]: Good catch — added the `"In Progress": 4` spaced-key case to the test. This locks in the behavior that `to_lowercase()` preserves spaces, which matches how Linear state names actually work (e.g. "In Progress" → "in progress"). Fixed in abb8552.

File: apps/symphony/AGENTS.md [hunk 2]
```diff
@@ -1,96 +1,425 @@
...
-- Domain types: `src/domain.rs`
-- Error types: `src/error.rs`
-- Decisions register: `.kata/DECISIONS.md` (append-only, read during planning)
-- Requirements: `.kata/REQUIREMENTS.md`
+- Full module layout and architecture overview
+- Reference to the Elixir implementation and SPEC.md (authoritative behavioural contract)
+- Module-to-source-file mapping table
+- Hard rules for maintaining spec parity
+- Build, test, and lint commands
+- Git workflow (worktrees, standby branches, commit conventions)
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/symphony/AGENTS.md:425]: The self-reference is real but the root cause is that `README.md` is a symlink to `AGENTS.md` (a repo-wide convention — see root `AGENTS.md`: "CLAUDE.md files in this repo are symlinks to AGENTS.md"). T02 wrote the operator README which overwrote through the symlink, merging the two files unintentionally.

The correct fix is to break the symlink and make README.md a standalone operator guide while preserving AGENTS.md as the dev reference — but that changes the repo-level convention and is outside the scope of S09's conformance work. Leaving this for a follow-up.

File: apps/symphony/.kata/activity/073-execute-task-M001-S09-T03.jsonl [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
...pe":"message","id":"f89a4517","parentId":"fa75b8c2","timestamp":"2026-03-19T19:20:42.295Z","message":{"role":"assistant","content":[{"type":"text","text":"All done. Here's a summary of what was completed:\n\n**Verification results:**\n- `cargo test` → **211 tests, 0 failures** (25+9+32+13+7+33+29+15+20+28 across 11 suites) ✓\n- `cargo clippy -- -D warnings` → **exits zero, no warnings** ✓\n\n**Bookkeeping updates:**\n- `REQUIREMENTS.md` — R013 status `active` → `validated`; proof note references 211-test total + two new S09 conformance tests; traceability row updated; Coverage Summary: active 3→2, validated 12→13 ✓\n- `STATE.md` — Phase: Complete, S09 checkbox done, R013 added to validated list, next action reflects M001 complete ✓\n- `S09-PLAN.md` — T03 marked `[x]` ✓\n- `T03-SUMMARY.md` written ✓\n\nTask T03 complete."}],"api":"anthropic-messages","provider":"anthropic","model":"claude-sonnet-4-6","usage":{"input":1,"output":244,"cacheRead":72704,"cacheWrite":319,"totalTokens":73268,"cost":{"input":0.000003,"output":0.00366,"cacheRead":0.0218112,"cacheWrite":0.00119625,"total":0.02667045}},"stopReason":"stop","timestamp":1773948036270,"responseId":"msg_01HgracrVqRvYH4L85ZNpKZ5"}}
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/symphony/.kata/activity/073-execute-task-M001-S09-T03.jsonl:45]: `.kata/activity/` files are raw JSONL session transcripts used for auto-mode retry diagnostics. They contain `bash` tool invocations which naturally include absolute paths from the machine that ran the session. Redacting these would break their diagnostic value — the paths are how you trace which commands ran and where.

These files are gitignored in normal repos (Kata's bootstrap adds `.kata/activity/` to `.gitignore`). They're in this PR because this is the project's own development repo where `.kata/` is tracked for planning artifacts. Not redacting.
